editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -5691,8 +5691,9 @@ void CanvasItemEditorViewport::_create_preview(const Vector<String> &files) cons
 }
 
 void CanvasItemEditorViewport::_remove_preview() {
+	canvas_item_editor->message = "";
+	canvas_item_editor->update_viewport();
```

Comment:
This should only happen when `message` is not empty.

Reply:
Yeah, just double checked. I could swear I was having issues with the old message staying when not doing an `update_viewport`.

Reply:
I mean, the message is staying in some cases (e.g. when you press Escape), you need to add a condition to avoid needlessly updating the viewport.

Reply:
No I mean you should keep `update_viewport()`, but wrap it in the condition (`message = ""` can be wrapped too).

Reply:
Ok I see. Should something similar need to happens in the `3D` viewport:

```
if (!message.is_empty()) {
	set_message("");
}
```

Asking as I took reference from here, this doesn't require an `update_viewport` equivalent as this is being done in `_draw`

Reply:
No, setting the message just assigns a String. There is no reason to make a new condition for it. In 2D the condition would already exist, so it can be reused.